feat(boxes): refactor box reward system with improved brawler handling

MinecAnton209 opened the pull request:
- Add TrophyRoadBrawlersInBoxesEnabled config option
- Improve brawler initialization in Player class
- Refactor LogicBoxData with better brawler data management
- Add error handling and logging for missing brawlers
- Update ClientAvatar to handle missing brawler data safely

MinecAnton209 committed:
feat(boxes): refactor box reward system with improved brawler handling

- Add TrophyRoadBrawlersInBoxesEnabled config option
- Improve brawler initialization in Player class
- Refactor LogicBoxData with better brawler data management
- Add error handling and logging for missing brawlers
- Update ClientAvatar to handle missing brawler data safely

8-bitHacc reviewed (commented):
Hey, thanks for your pull request.
Even though most of your changes are good, I feel like the code behind it isn't the best, so I will not be merging these changes. Your code _**might**_ help in the implementations of these features in the future.

MinecAnton209 committed:
feat(boxes): Restore original box chances

- Restored original drop chances for Small, Big, and Mega Boxes in LogicBoxData.
- Improved player data handling in Logic/Player.py to support more dynamic features.

8-bitHacc commented:
I'd say maybe not touching box code for now because it will be rewritten soon.

MinecAnton209 commented:
> I'd say maybe not touching box code for now because it will be rewritten soon.

Understood, thank you. I'll leave this version for local use for now, maybe some ideas or logic will be useful for a future update.
